src/object/srv_ec.c

```diff
@@ -93,39 +122,77 @@ obj_ec_rw_req_split(struct obj_rw_in *orw, struct obj_ec_split_req **split_req)
 	req->osr_start_shard = start_shard;
 
 	for (i = 0; i < tgt_nr; i++) {
-		tgt_idx = fw_tgts[i].st_shard - start_shard;
-		D_ASSERT(tgt_idx < tgt_max_idx);
+		if (tgt_map != NULL) {
+			if (!obj_ec_is_valid_tgt(tgt_map, map_size,
+						 tgts[i].st_tgt_id, &tgt_idx))
+				continue;
+
+			D_ASSERT(tgt_idx >= start_shard);
+
+			tgt_idx -= start_shard;
+			if (tgt_max_idx < tgt_idx)
+				tgt_max_idx = tgt_idx;
+		} else {
+			tgt_idx = tgts[i].st_shard - start_shard;
+			D_ASSERT(tgt_idx < tgt_max_idx);
+		}
+
 		setbit(tgt_bit_map, tgt_idx);
+		count++;
+	}
+
+	if (tgt_map != NULL) {
+		D_ASSERT(count == map_size);
+
+		if (obj_ec_is_valid_tgt(tgt_map, map_size,
+					tgts[0].st_tgt_id, &tgt_idx))
+			tgt_idx -= start_shard;
+		else
+			/* If the leader (tgt[0]) is not in current EC RDG,
+			 * then set 'tgt_idx' as zero to select 'tgt_oiod'.
+			 * Related modification will not be executed on the
+			 * leader.
+			 */
+			tgt_idx = 0;
```

Comment:
Then I may always elect the parity node index?

Reply:
the DTX leader tgts[0] possibly has nothing to do with this EC obj update's RDG/tgts, here seems fine if it can ensure "Related modification will not be executed on the leader".  But may need to avoid the assertion in below L170 and handle the case of NULL tgt_oiod. just FYI.